zerver/views/home.py

```diff
@@ -249,6 +249,8 @@ def home_real(request: HttpRequest) -> HttpResponse:
     if user_profile.is_guest:
         show_invites = False
         show_add_streams = False
+    if only_auth_enabled(['LDAP'], user_profile.realm) and not user_profile.realm.invite_required:
+        show_invites = False
```

Comment:
Let's add a comment explaining the reasoning behind this block, since it's a bit subtle.